lib/http.c

```diff
@@ -4250,13 +4250,13 @@ CURLcode Curl_http_readwrite_headers(struct Curl_easy *data,
 
         /* There can only be a 4th response code digit stored in 'digit4' if
            all the other fields were parsed and stored first, so nc is 5 when
-           digit4 is not -1 */
-        else if(digit4 != -1) {
+           digit4 a digit */
```

Comment:
this comment needs to be corrected as well, digit4 may not be a digit. maybe you should call it separator2, optional separator. it is required to be a space but we never enforced that i don't think

Reply:
The line doesn't look like that anymore, and I don't see any problem with the comment as it is now. Can you?

Reply:
The line I see in the code review, just refreshed, ends with "nc is 5 when digit4 a digit" but actually nc is 5 when digit4 has received any character not necessarily a digit.

Reply:
The comment doesn't say that though. It says that when digit4 is a digit, nc is 5. I think that's correct. nc is also 5 when digit4 is space, which the comment doesn't say but the line of code the comment refers to is about the digit check.

Reply:
> nc is also 5 when digit4 is space, which the comment doesn't say but the line of code the comment refers to is about the digit check.

ok fair enough